pkg/graveler/graveler.go

```diff
@@ -1286,6 +1295,66 @@ func (g *KVGraveler) SaveGarbageCollectionCommits(ctx context.Context, repositor
 	}, err
 }
 
+type GCUncommittedMark struct {
+	BranchID BranchID `json:"branch"`
+	Key      Key      `json:"key"`
+}
+
+func (g *KVGraveler) GetGarbageCollectionUncommitted(ctx context.Context, repository *RepositoryRecord, mark GCUncommittedMark) ([]*ValueRecord, *GCUncommittedMark, error) {
+	var entries []*ValueRecord
+	bItr, err := g.ListBranches(ctx, repository)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer bItr.Close()
+	bItr.SeekGE(mark.BranchID)
+
+	count := 0
+	for bItr.Next() {
+		b := bItr.Value()
+		itr, err := g.listStagingArea(ctx, b.Branch)
+		if err != nil {
+			bItr.Close()
+			return nil, nil, err
+		}
+		itr.SeekGE(mark.Key)
```

Comment:
That's a bug, not every entry iterator should `SeekGE`. Just the first one.

Reply:
Also the `mark.Key` is valid only if the branch equals to `mark.Branch`. Imagine the mark is `branch1,key200`, and the `branch1` was deleted. You will now start from `branch2,key200` although you didn't cover prior keys in `branch2` at all!

Reply:
😨